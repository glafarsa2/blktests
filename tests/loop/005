#!/bin/bash
#
# Test the LOOP_GET_STATUS{,64} error case where the passed argument is NULL.
# Regression test for patch "loop: fix LOOP_GET_STATUS lock imbalance".
#
# Copyright (C) 2018 Omar Sandoval
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

DESCRIPTION="call LOOP_GET_STATUS{,64} with a NULL arg"

QUICK=1

requires() {
	_have_src_program loop_get_status_null
}

test() {
	local loop_dev

	echo "Running ${TEST_NAME}"

	truncate -s 1M "$TMPDIR/file"
	if ! loop_dev="$(losetup -f --show "$TMPDIR/file")"; then
		return 1
	fi

	"$SRCDIR/loop_get_status_null" "$loop_dev"
	"$SRCDIR/loop_get_status_null" "$loop_dev" 64

	losetup -d "$loop_dev"

	echo "Test complete"
}
