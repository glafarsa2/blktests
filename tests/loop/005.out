Running loop/005
Got EINVAL
Got EINVAL
Test complete
